### ISSUE TITLE
[Data Loss Bug] Obsidian Sync Rejects Changes When Plugin Enabled

### ISSUE THREAD

**Issue by yuboWw** [Developer-Mike/obsidian-advanced-canvas#116]
this is a new issue i have run into lately.
i have troubleshot and verified that it is the Advanced Canvas plugin that is causing this issue.

# bug reproducible description

- Obsidian Sync is enabled across the devices
- a canvas file is available on mobile and on desktop
- both devices are "Fully Synced" on the Obsidian Sync status
- quit the Obsidian app on mobile
- make changes on desktop on that canvas file
- launch Obsidian app on mobile

🐛 Obsidian Sync log will display "rejected server change" on that canvas file, and then it will proceed to upload the local version of that canvas file on mobile (before all the changes were made on desktop) to the server, resulting in the newest version on desktop being replaced by the old version from mobile, and therefore a loss of work progress (reversible by Undo if the canvas file is opened on desktop this entire time)

# troubleshooting steps performed

- follow the aforementioned steps, with Advanced Canvas enabled, the issue occurs every time
- follow the aforementioned steps, with Advanced Canvas disabled, the issue does not occur

it was also noted that the Obsidian Sync's syncing and indexing speed is much faster with the plugin disabled. this might be related.

**Comment by Developer-Mike** [comment]
Sounds like an important issue. Can you try using [this](https://github.com/Developer-Mike/obsidian-advanced-canvas/issues/104#issuecomment-2457952807) modification?

**Comment by Developer-Mike** [comment]
Any updates?

**Comment by yuboWw** [comment]
hi, mike @Developer-Mike, 

no, i'm sorry. 

since then, i have temporarily discontinued using Advanced Canvas as part of the normative operation of my system, due to this high instability and insecurity.

still appreciate this fantastic addition to the default Canvas and your work, but my current life phase doesn't allow much time on tweaking and tinkering Obsidian.

**Comment by Developer-Mike** [comment]
@yuboWw I've just bought sync to debug this issue, but I can't reproduce it. Can you confirm that it still occurs with the newest version of AC?

**Comment by Developer-Mike** [comment]
@maxrho Can you explain your issue (#140) in more detail. Can't reproduce it either.

**Comment by yuboWw** [comment]
> [@yuboWw](https://github.com/yuboWw) I've just bought sync to debug this issue, but I can't reproduce it. Can you confirm that it still occurs with the newest version of AC?

appreciate it!
i can't confirm, i haven't been using AC since. however, the OP description was the full picture of the situation i was dealing with using the latest version of AC at that time.

**Comment by Developer-Mike** [comment]
Very likely related to #173. The aforementioned issue will get fixed in version `3.9.1`. I mark this as duplicate for now but please reopen it if this issue persists in version `3.9.1`.

**Comment by yuboWw** [comment]
is #70 also related to #173 you think? it seems that the 173 guy has found the root problem

**Comment by Developer-Mike** [comment]
I'm not so sure if #70 is also related. I have difficulties understanding the potential problem at all. I wasn't able to reproduce it in any way (except sync conflicts that also occur without the plugin). But it may have gotten resolved, too.

**Comment by yuboWw** [comment]
i'm not familiar with the way Obsidian Sync works, but if it relies on the Last Modified Date of notes to function, then #173 might be it

**Comment by Developer-Mike** [comment]
Yes, it apparently does: https://discord.com/channels/686053708261228577/1342818357409349642/1342921744297496628

**Comment by yuboWw** [comment]
oh, ok
good to know
